src/api/Feed.js

```diff
@@ -0,0 +1,1022 @@
+     * @param {string} dueAt - The optional date the task is due
+     * @return {void}
+     */
+    updateTask = (
+        file: BoxItem,
+        taskId: string,
+        message: string,
+        successCallback: (task: Task) => void = noop,
+        errorCallback: (e: $AxiosXHR<any>) => void = noop,
+        dueAt?: string
+    ) => {
+        if (!file.id) {
+            throw getBadItemError();
+        }
+
+        this.id = file.id;
+        this.updateFeedItem({ isPending: true }, taskId);
+        this.tasksAPI = new TasksAPI(this.options);
+        this.tasksAPI.updateTask({
+            file,
+            taskId,
+            message,
+            dueAt,
+            successCallback: (task: Task) => {
+                this.updateTaskSuccessCallback(task, successCallback);
+            },
+            errorCallback: (e: $AxiosXHR<any>) => {
+                this.updateTaskErrorCallback(e, errorCallback);
+            }
+        });
+    };
+
+    /**
+     * Update task error callback
+     *
+     * @param {Object} e - Axios error
+     * @param {Function} errorCallback - The error callback
+     * @return {void}
+     */
+    updateTaskErrorCallback = (e: $AxiosXHR<any>, errorCallback: Function) => {
+        this.errorCallback(e);
+        if (!this.isDestroyed()) {
+            errorCallback(e);
+        }
+    };
+
+    /**
+     * Update task success callback
+     *
+     * @param {Object} task - The updated task
+     * @param {Function} successCallback - The success callback
+     * @return {void}
+     */
+    updateTaskSuccessCallback = (task: Task, successCallback: Function) => {
+        const updates = omit(task, TASK_ASSIGNMENT_COLLECTION);
+        this.updateFeedItem(
+            {
+                ...updates,
+                isPending: false
+            },
+            task.id
+        );
+        if (!this.isDestroyed()) {
+            successCallback(task);
+        }
+    };
+
+    /**
+     * Deletes a comment.
+     *
+     * @param {BoxItem} file - The file to which the comment belongs to
+     * @param {string} commentId - Comment ID
+     * @param {BoxItemPermission} permissions - Permissions for the comment
+     * @param {Function} successCallback - the function which will be called on success
+     * @param {Function} errorCallback - the function which will be called on error     *
+     * @return {void}
+     */
+    deleteComment = (
+        file: BoxItem,
+        commentId: string,
+        permissions: BoxItemPermission,
+        successCallback: Function,
+        errorCallback: Function
+    ): void => {
+        this.commentsAPI = new CommentsAPI(this.options);
+        if (!file.id) {
+            throw getBadItemError();
+        }
+
+        this.id = file.id;
+        this.updateFeedItem({ isPending: true }, commentId);
+
+        this.commentsAPI.deleteComment({
+            file,
+            commentId,
+            permissions,
+            successCallback: () => {
```

Comment:
I know we like to use arrow functions, but do we get anything out of it that couldn't be done via `.bind()`? In the end we're making a new function object that's bound to the calling scope.

ie) `successCallback: this.deleteFeedItem.bind(this, commentId, successCallback)`

Reply:
This comment also refers to other places where we use an arrow function, in this manner, with no arguments

Reply:
yea that would be cleaner 👍